### ISSUE TITLE
[BUG] Intermittent 403 authorizationfailure when performing operations with a BlobServiceClient

### ISSUE THREAD

**Issue by ADH-LukeBollam** [Azure/azure-sdk-for-net#45639]
### Library name and version

Azure.Storage.Blobs 12.20.0

### Describe the bug

I get an intermittent 403 authorization failure when performing block blob operations such as delete, put and create container. I'm synchronizing many files to a new storage account and it can run fine for hours, and then randomly will fail with a 403. Rerunning immediately afterwards will successfully perform the exact same operations.

- I use a single BlobServiceClient throughout.
- I connect to the storage account with a connection string. 
- I have added my personal IP address to the storage whitelist. Access is also allowed through Service Endpoints of Virtual Network in Azure, but my PC is not part of that.

I haven't been able to find any information from docs about what could be causing me to intermittently become unauthorized. Checking in Azure, I can see some Authorization Failures in the metrics but have been unable to get any detailed logs about why from the Log Analytics Workspace.

### Expected behavior

Operations can proceed without authorisation errors. If there is an authorization error, details are available so I can try to fix it.

### Actual behavior

Random authorization errors.

### Reproduction Steps

In Azure
1. Create a storage account and add your personal PC IP address to the networking whitelist

In your code
1. Get a BlobServiceClient from DI using a connection string
2. Create containers if they dont exist
3. Use it to check that blobs exist
4. Delete old copies of blobs
5. Write new copies of blobs

Wait random amount of time from 5 minutes to 2 hours for a 403 response

### Environment

.Net Core 8 Console Application
Visual Studio 17.10.4

**Comment by nickliu-msft** [comment]
Hi @ADH-LukeBollam, would you be able to provide the exact 403 error message?

Also can you upgrade to the newest version (12.21.1) and see if the issue still persists?

**Comment by ADH-LukeBollam** [comment]
I noticed the issue would only happen while performing lots of async read operations with spotty internet. I ended up working around it by just running tasks synchronously and running the script in Azure in the same region as the storage.

Sorry I can't provide any more debug information right now, but my guess would be that the bad internet caused too many async operations to queue up at once, and I was exceeding the limit of open connections.
